src/validator/data_model.py

```
@@ -1117,20 +1117,63 @@ def validate_non_empty(cls, v: str, info: "ValidationInfo") -> str:
+            result.append(f"For: {day_type}")
+            result.extend(time_data)
+        return result
+
+    @classmethod
+    def _validate_until(cls, data: list) -> list[str]:
+        result = []
+        for i, item in enumerate(data):
+            time = item["Until"]["Time"]
+            value = float(item["Until"]["Value"])
+            if i == len(data) - 1 and time != "24:00":
+                raise ValueError(f"Last time entry must be 24:00, but got {time}")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)